src/core/algorithms/ind/spider/attribute.h

```diff
@@ -84,6 +84,12 @@ class AINDAttribute final : public details::Attribute {
 
     /// get referenced attribute indices
     std::vector<AttributeIndex> GetRefIds(config::ErrorType max_error) const;
+
+    /// get error threshold
+    config::ErrorType GetError(AttributeIndex ref_id) const {
+        auto const dep_count = static_cast<config::ErrorType>(occurrences_[id_]);
+        return 1 - occurrences_[ref_id] / dep_count;
+    }
```

Comment:
I don't know the logic behind filling `occurrences_` but when I see a variable with name `count`something I assume it _can_ be zero, in which case we will be dividing by it. `dep_count` cannot be zero, right?

Reply:
Yes, there is a guarantee that the `dep_count` is not zero. This is actually the number of unique values ​​for the dependent attribute (`this->id_`). And we know that at least one value exists, since we do not process empty tables.